src/EUTelGeometryTelescopeGeoDescription.cpp

```diff
@@ -883,14 +941,105 @@ const TGeoHMatrix* EUTelGeometryTelescopeGeoDescription::getHMatrix( const doubl
  * 
  * @return radiation length in units of X0
  */
-float EUTelGeometryTelescopeGeoDescription::findRadLengthIntegral( const double globalPosStart[], const double globalPosFinish[], bool skipBoundaryPonitsVolumes ) {
 
+float EUTelGeometryTelescopeGeoDescription::findRad( const double globalPosStart[], const double globalPosFinish[], bool skipBoundaryPonitsVolumes, 	std::map< const int, double> &sensors, 	std::map< const int, double> &air ){
```

Comment:
I agree with you here. However at the moment can we leave this here since I am unsure what moving this to another class will do to identifying volumes.